src/codeu/chat/client/ClientUser.java

```diff
@@ -88,16 +89,20 @@ public void showCurrent() {
     printUser(current);
   }
 
-  public void addUser(String name) {
-    final boolean validInputs = isValidName(name);
+  public void addUser(String name, String password) {
+    
+    final boolean validInputsName = isValidName(name);
+    final boolean validInputsPass = isValidName(password);
+    //TODO: Write isValidPassword(String password); method
+    //validInputs = isValidPassword(password);
 
-    final User user = (validInputs) ? controller.newUser(name) : null;
+    final User user = (validInputsName && validInputsPass) ? controller.newUser(name, password) : null;
 
     if (user == null) {
       System.out.format("Error: user not created - %s.\n",
-          (validInputs) ? "server failure" : "bad input value");
+          (validInputsName && validInputsPass) ? "server failure" : "bad input value");
     } else {
-      LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
+      LOG.info("New user complete, Name= \"%s\" UUID=%s Password= \"%s\"", user.name, user.id, user.password);
```

Comment:
It's worth thinking about the dangers in logging the password like this. Even if you switched to an encrypted approach later on, you're now logging plaintext passwords.